pkg/apiserver/auditing/backend.go

```diff
@@ -107,3 +109,27 @@ func (b *Backend) worker() {
 		go send(event)
 	}
 }
+
+func (b *Backend) eventToBytes(event *v1alpha1.EventList) ([]byte, error) {
+
+	if bs, err := json.Marshal(event); err == nil {
+		return bs, nil
+	}
+
+	// Normally, the serialization failure is caused by the failure of RequestObject or ResponseObject serialization.
+	// To ensure the integrity of the auditing event to the greatest extent,
+	// it is necessary to delete RequestObject or ResponseObject and and then try to serialize again.
+	if event.Items[0].RequestObject != nil {
```

Comment:
We need to come up with a better solution, in the worst case, you have to marshal four times to get the result, which is very time-consuming.

Reply:
Request body of all current apis are json formate, only response body of some few apis are not json formate, so i will delete the serializable detection when serialization failure，directly set ResponseObject to null, and then try serialization again. This is a compromise between data integrity and performance. I will consider how to optimize it here later.